spec.md

```diff
@@ -329,6 +331,12 @@ The process remains unchanged for chunked upload, except that the post request M
 Content-Length: 0
 ```
 
+When pushing a blob with a digest algorithm other than `sha256`, the post request SHOULD include the `digest-algorithm` parameter:
```

Comment:
Why do we need this new query param? We already send the expected digest (including the algorithm) as a mandatory parameter when closing the blob upload.

Reply:
Two reasons I can think of. It's a courtesy to registries that compute the digest as the blob is received and don't want to reread the blob to recompute it with a new algorithm. It's also a way for clients to get a fast failure if they request an algorithm not supported by the registry, before spending the time and bandwidth to upload the blob.

Reply:
Any particular reason this is a query param instead of a header? so API endpoint defs don't change (?)
What are our guidelines regarding what gets set as a header and what as a query param.

Reply:
We use query params in a lot of other parts of the spec. Is there a reason this should be a header?

Reply:
> In general, older registries will reject in the PUT step, and we cannot assume that registries are implementing streamed hashing.

Correct, the failure condition on older registries requires the blob to be pushed first. This gives newer registries the ability to improve performance of the blob push.

Reply:
same here... the client may include sha256 to ensure it is supported, this for future use cases presuming there will eventually be replacement(s)